src/lambda/wizards/samDeployWizard.ts

```diff
@@ -702,7 +722,17 @@ export class SamDeployWizard extends MultiStepWizard<SamDeployWizardResponse> {
     private readonly TEMPLATE: WizardStep = async () => {
         // set steps back to 0 since the next step determines if additional steps are needed
         this.context.additionalSteps = 0
-        this.response.template = await this.context.promptUserForSamTemplate(this.response.template)
+        if (this.samTemplateFile && this.response.template) {
+            // This state means:
+            //   1. wizard was started from a context-menu (`this.samTemplateFile` was set)
+            //   2. user canceled the REGION step.
+            // => User wants to exit the wizard.
+            return WIZARD_TERMINATE
+        } else if (this.samTemplateFile) {
+            this.response.template = this.samTemplateFile
```

Comment:
Nit: this case will mean shrinking the wizard's totalSteps by 1 since this step has essentially been pruned.

Reply:
I tried, but there are things like this: https://github.com/aws/aws-toolkit-vscode/blob/7629a1e77495014c0eef70a9a4974d38e8082ac5/src/lambda/wizards/samDeployWizard.ts#L637

Changing the step count is very fragile. I will convert this wizard to @JadenSimon 's new wizard framework later.